tests/Tests.AzureAppConfiguration/KeyVaultReferenceTests.cs

```diff
@@ -296,7 +296,6 @@ public void ServerRequestIsMadeWhenDefaultCredentialIsSet()
             });
 
             Assert.NotNull(ex.InnerException);
-            Assert.IsType<AuthenticationFailedException>(ex.InnerException);
```

Comment:
Didn't we have this assertion to verify this line https://github.com/Azure/AppConfiguration-DotnetProvider/blob/c3ed5a3f58096db2bf781724fd07ec15ac3fd9c9/src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureKeyVaultReference/AzureKeyVaultKeyValueAdapter.cs#L55 ?

Reply:
I do see that we added it in the same PR that we added that line: https://github.com/Azure/AppConfiguration-DotnetProvider/commit/fbde2ecd1a0262675a1c677f322a1a0fec26c1da#diff-1a142c991b5e50180b6e1403412391179ffb37afdccedaaba3567696eeb82efa

Reply:
I guess all we can really ensure is that `KeyVaultReferenceException` is the top level exception, so should be fine. Just wanted to make sure that we didn't need this check since I remember we had some discussion about this line specifically in the past.

Reply:
As long as the actual exception is wrapped in `KeyVaultReferenceException`, we should be good.